File: report/src/main/java/com/vmlens/report/input/run/objecthashcodemap/MultipleThreadIndices.java
package com.vmlens.report.input.run.objecthashcodemap;

public class MultipleThreadIndices implements ThreadIndices {

    private final int code;

    public MultipleThreadIndices(int code) {
        this.code = code;
    }

    @Override
    public ThreadIndices addThreadIndex(int newThreadIndex,ObjectHashCodeMap objectHashCodeMap) {
        return this;
    }

    @Override
    public String asUiString() {
        return "" + code;
    }
}

File: sync-bug/src/main/scala/com/anarsoft/race/detection/report/element/runelementtype/objecthashcodemap/MultipleThreadIndices.scala
package com.anarsoft.race.detection.report.element.runelementtype.objecthashcodemap

class MultipleThreadIndices(val code: Int) extends ThreadIndices {
  override def addThreadIndex(newThreadIndex: Int, objectHashCodeMap: ObjectHashCodeMap): ThreadIndices = this

  override def asUiString: String = "" + code

  def isSingleThreaded: Boolean = false;

  def id : Option[Int] = Some(code);
}
